programs/account-compression/src/state/batched_queue.rs

```diff
@@ -271,7 +267,6 @@ impl ZeroCopyBatchedQueueAccount {
                     .ok_or(AccountCompressionErrorCode::InclusionProofByIndexFailed)?;
 
                 if element == value {
-                    *element = [0; 32];
```

Comment:
unfortunately we need to revert this because it leads to some issues onchain.
This means that we can't get the leaves for the batch append circuit inputs from the batch Merkle tree account value array because some values have been zeroed out.
Instead we need to get the data from the indexer. I think it makes sense to add a `get_queue_elements` method to the `Indexer` trait like I have for the batch address tree [see here](https://github.com/Lightprotocol/light-protocol/pull/1365/commits/4df70ae882f9cfe8e93992b4cf4e9522cbf8d489#diff-42faea6ff4f47f30e026b0ead2044d53a053289517ed9aa60ae996efaa87f30dR76-R91).